src/evo/dmn_types.h

```diff
@@ -51,8 +68,13 @@ constexpr auto HighPerformance = mntype_struct{
     switch (type) {
         case MnType::Regular: return dmn_types::Regular;
         case MnType::HighPerformance: return dmn_types::HighPerformance;
-        default: assert(false);
+        default: return dmn_types::Invalid;
     }
 }
 
+[[nodiscard]] constexpr const bool IsValidMnType(MnType type)
+{
+    return GetMnType(type) != dmn_types::Invalid;
```

Comment:
I think better to iomplement it like `return (static_cast<uint16_t>(type) > MnType::LAST);`

Reply:
And define LAST like:
...
HighPerformance = 1,
LAST = HighPerformance,
Invalid = ....